### PR TITLE
Integrate My Profile page

### PR DESCRIPTION
**Pull request by filipsobol**
### Short Description and Why It's Useful
This PR integrates "Personal Data" and "Password Reset" forms in "My Profile" page with methods exposed by `useUser`.

Additionally it:
- Adds new `getEmailAddress` getter in `userGetters`,
- Makes checkbox `I want to create an account` required in registration form,
- Improves error flow, so they can be properly catched in the template and in the future, displayed nicely to use user.

### Upgrade Notes and Changelog
- [ ] No upgrade steps required (100% backward compatibility and no breaking changes)
- [X] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

### CODE REVIEW

File: packages/commercetools/composables/src/useUser/authenticate.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export const authenticate = async (userData: UserData, fn) => {
     const userResponse = await fn(userData);
     return userResponse.data.user;
   } catch (err) {
-    console.error(err.graphQLErrors ? err.graphQLErrors[0].message : err);
+    err.message = err?.graphQLErrors?.[0]?.message || err.message;
+    console.error(err.message);
```
Comment (filrak) [inline packages/commercetools/composables/src/useUser/authenticate.ts:11]: **change** use Logger

File: packages/commercetools/theme/components/profile/PasswordResetForm.vue [hunk 2]
```diff
@@ -0,0 +1,126 @@
+<template>
```
Comment (filrak) [inline packages/commercetools/theme/components/profile/PasswordResetForm.vue:1]: **suggestion** In the project a convention is to name folders in `components` that group components specific to a certain page in a dir with a name of that page (eg `Checkout.vue` - page, `checkout` - dir in components) and other components either in a root or grouped if they're all about a specific feature. These components are part of `MyAccount` so lets rename that dir accordingly :)

File: packages/commercetools/theme/components/profile/PasswordResetForm.vue [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+  components: {
+    SfInput,
+    SfButton,
+    ValidationProvider,
+    ValidationObserver
+  },
+
+  setup(_, { emit }) {
+    const clearForm = () => ({
```
Comment (filrak) [inline packages/commercetools/theme/components/profile/PasswordResetForm.vue:63]: **question** isn't this implying that we're clearing the `form` not returning its default values?

**Review by filrak — CHANGES_REQUESTED** [review]
Nice job! Everything is much cleaner ;) I've added few DX-related suggestions

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**